### PR TITLE
[HOPSWORKS-1422] Custom metadata to featuregroups

### PR DESCRIPTION
**Pull request by maismail**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [x] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1422

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
tests: https://github.com/logicalclocks/hops-testing/pull/546

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsController.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import javax.ejb.Stateless;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+
+@Stateless(name = "XAttrsController")
+public class XAttrsController {
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsController.java]: @maismail set transactions to `never` https://github.com/logicalclocks/hopsworks/blob/v1.0.0/hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/JobController.java#L70

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    }
+    
+    xattrsController.addXAttr(project, user, path, xattrName,
+        metaObj);
+    
+    UriBuilder builder = uriInfo.getAbsolutePathBuilder();
+    if(status == Response.Status.CREATED) {
+      return Response.created(builder.build()).build();
+    } else {
+      return Response.ok(builder.build()).build();
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java]: @maismail PUT endpoint should return the created entity https://github.com/logicalclocks/hopsworks/blob/v1.0.0/hopsworks-api/src/main/java/io/hops/hopsworks/api/jobs/JobsResource.java#L196

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  private Project project;
+  
+  public void setProject(Integer projectId) {
+    this.project = projectFacade.find(projectId);
+  }
+  
+  
+  @ApiOperation( value = "Create or Update an extended attribute for a path.")
+  @PUT
+  @Path("{path: .+}")
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java]: @ErmiasG Would that fail if we create a dataset with name `datasets`? As in the new datsets API we do `/project/{id}/datasets/{path: .+}` ?

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsController.java (hunk 1 above)
Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsController.java]: Done.

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java (hunk 2 above)
Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java]: Done.

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java (hunk 3 above)
Comment (ErmiasG) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/metadata/XAttrsResource.java]: it is under @Path("{projectId}/xattrs"), so it should resolve to /project/{id}/xattrs/datasets

**Pull request merged**